src/main/java/io/confluent/kafkarest/entities/TopicPartitionOffsetMetadata.java

```diff
@@ -97,7 +97,7 @@ public boolean equals(Object o) {
     if (offset != that.offset) {
       return false;
     }
-    if (metadata != that.metadata) {
```

Comment:
Well spotted! But I think `Object.equals` will just do a reference comparison. 
Can you update metadata value comparison to be the same as the topic value comparison below? thanks! 

https://stackoverflow.com/questions/13001971/using-the-equals-method-with-string-and-object-in-java